river/command/rule.zig

```diff
@@ -100,29 +111,28 @@ pub fn ruleDel(_: *Seat, args: []const [:0]const u8, _: *?[]const u8) Error!void
     if (result.args.len < 1) return Error.NotEnoughArguments;
 
     const action = std.meta.stringToEnum(Action, result.args[0]) orelse return Error.UnknownOption;
-    const app_id_glob = result.flags.@"app-id" orelse "*";
-    const title_glob = result.flags.title orelse "*";
 
+    const rule = .{
+        .app_id_glob = result.flags.@"app-id" orelse "*",
+        .title_glob = result.flags.title orelse "*",
+    };
     switch (action) {
         .float, .@"no-float" => {
-            server.config.float_rules.del(.{
-                .app_id_glob = app_id_glob,
-                .title_glob = title_glob,
-            });
+            server.config.float_rules.del(rule);
         },
         .ssd, .csd => {
-            server.config.ssd_rules.del(.{
-                .app_id_glob = app_id_glob,
-                .title_glob = title_glob,
-            });
+            server.config.ssd_rules.del(rule);
             apply_ssd_rules();
             server.root.applyPending();
         },
         .tag => {
-            server.config.tag_rules.del(.{
-                .app_id_glob = app_id_glob,
-                .title_glob = title_glob,
-            });
+            server.config.tag_rules.del(rule);
+        },
+        .output => {
+            if (server.config.output_rules.find(rule)) |output_rule| {
+                util.gpa.free(output_rule.value);
+            }
+            server.config.output_rules.del(rule);
```

Comment:
Instead of adding a separate `find()` function and searching twice, why don't we make `del()` return the deleted rule's value as an optional (`?T`)?